### PR TITLE
feat(TDI-39092): Implement cross module test coverage report for sonar

### PR DESCRIPTION
**Pull request by iistomin-gl**
**What is the current behavior?** (You can also link to an open issue here)
- For multi-module project (component), Sonar provides test coverage report for single module, f.x. classes covered by integration tests which are in separate 'integration' module are not counted in report (definition, runtime etc.)

**What is the new behavior?**
- For multi-module project (component), Sonar provides cross module, aggregated test coverage report for each module, f.x. classes covered by integration tests which are in separate 'integration' module are counted in report (definition, runtime etc.)

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [X] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [X] Test coverage

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

**Other information**:
JIRA issue: https://jira.talendforge.org/browse/TDI-39092

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/764/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/764/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/766/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/766/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/761/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/761/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/770/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/770/)

File: components/components-elasticsearch/elasticsearch-integration/pom.xml [hunk 1]
```diff
@@ -18,6 +18,7 @@
     <properties>
         <!-- skip the deployment for integration tests, not necessary -->
         <maven.deploy.skip>true</maven.deploy.skip>
+        <coverage.sonar.reportRoot>${project.basedir}/..</coverage.sonar.reportRoot>
```
Comment (gonchar-ivan) [inline components/components-elasticsearch/elasticsearch-integration/pom.xml]: Doesn't this module require `<argLine>${jacoco.agent.it.argLine}</argLine>` ?
Could you please explain about jacoco.agent.it.argLine property. I didn't get what it does

Comment (iistomin-gl) [inline components/components-elasticsearch/elasticsearch-integration/pom.xml]: If a pom.xml in some module doesn't override/customize argLine configuration element then it is not required to pass jacoco.agent.it.argLine property, by default component-parent/pom.xml does this.

Comment (iistomin-gl) [inline components/components-elasticsearch/elasticsearch-integration/pom.xml]: Regarding jacoco.agent.it.argLine property ... it contains jacoco specific command line arguments which are required to perform instrumentation of classes and to collect test coverage data.

Before my changes, a module overriding/customizing argLine in failsafe-plugin was loosing jacoco specific args and therefore test coverage data was not collected.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/801/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/801/)

**Pull request merged**